src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/ContextClasses.cs

```diff
@@ -118,4 +118,23 @@ internal partial class DictionaryTypeContext : JsonSerializerContext { }
 
     [JsonSerializable(typeof(JsonMessage))]
     public partial class PublicContext : JsonSerializerContext { }
+
+    [JsonSerializable(typeof(JsonMessage))]
+    public partial class GenericContext<T> : JsonSerializerContext { }
+
+    public partial class ContextGenericContainer<T>
+    {
+        [JsonSerializable(typeof(JsonMessage))]
+        public partial class NestedInGenericContainerContext : JsonSerializerContext { }
```

Comment:
consider similar test but one more nested type around, also another generic nested

Reply:
note if you're having issue with getting that to work treat this as optional I'm ok with just filing an issue and leaving as is, this PR is already huge improvement over previous state - I'm more interested in figuring out if we're fixing all cases or partially

Reply:
also consider `[JsonSerializable(typeof(Foo<T>))]` if that is even allowed

Reply:
I am afraid I can't do this quickly. In a couple of hours I will go to a countryside =) I'll be back at 18th july and can continue.
If the PR is good enough feel free to apply =)